gapis/api/vulkan/find_issues.go

```diff
@@ -55,17 +88,193 @@ func (t *findIssues) reportTo(r replay.Result) { t.res = append(t.res, r) }
 
 func (t *findIssues) Transform(ctx context.Context, id api.CmdID, cmd api.Cmd, out transform.Writer) {
 	ctx = log.Enter(ctx, "findIssues")
+	cb := CommandBuilder{Thread: cmd.Thread(), Arena: t.state.Arena}
+
 	mutateErr := cmd.Mutate(ctx, id, t.state, nil /* no builder */)
 	if mutateErr != nil {
 		// Ignore since downstream transform layers can only consume valid commands
 		return
 	}
-	out.MutateAndWrite(ctx, id, cmd)
+
+	s := out.State()
+	l := s.MemoryLayout
+
+	// Before an instance is to be destroyed, check if it has debug report callback
+	// created by us, if so, destory the call back handle.
+	if di, ok := cmd.(*VkDestroyInstance); ok {
+		inst := di.Instance()
+		if ch, ok := t.reportCallbacks[inst]; ok {
+			out.MutateAndWrite(ctx, api.CmdNoID, cb.ReplayDestroyVkDebugReportCallback(inst, ch))
+		}
+	}
+
+	switch cmd := cmd.(type) {
+	// Modify the vkCreateInstance to remove any validation layers first, and
+	// insert the individual validation layers in order. This is because Android
+	// does not support the meta layer, and the order does matter.Also enable the
+	// VK_EXT_debug_report extension.
+	case *VkCreateInstance:
+		cmd.Extras().Observations().ApplyReads(s.Memory.ApplicationPool())
+		info := cmd.PCreateInfo().MustRead(ctx, cmd, s, nil)
+		layerCount := info.EnabledLayerCount()
+		layers := []Charᶜᵖ{}
+		for _, l := range info.PpEnabledLayerNames().Slice(0, uint64(layerCount), l).MustRead(ctx, cmd, s, nil) {
+			if !isValidationLayer(l.String()) {
+				layers = append(layers, l)
+			}
+		}
+
+		validationLayersData := []api.AllocResult{}
+		for _, v := range validationLayers {
+			d := s.AllocDataOrPanic(ctx, v)
+			validationLayersData = append(validationLayersData, d)
+			layers = append(layers, NewCharᶜᵖ(d.Ptr()))
+		}
+		layersData := s.AllocDataOrPanic(ctx, layers)
+
+		extCount := info.EnabledExtensionCount()
+		exts := info.PpEnabledExtensionNames().Slice(0, uint64(extCount), l).MustRead(ctx, cmd, s, nil)
+		var debugReportExtNameData api.AllocResult
+		hasDebugReport := false
+		for _, e := range exts {
+			if e.String() == debugReportExtension {
+				hasDebugReport = true
+			}
+		}
+		if !hasDebugReport {
+			debugReportExtNameData = s.AllocDataOrPanic(ctx, debugReportExtension)
+			exts = append(exts, NewCharᶜᵖ(debugReportExtNameData.Ptr()))
+		}
+		extsData := s.AllocDataOrPanic(ctx, exts)
+
+		info.SetEnabledLayerCount(uint32(len(layers)))
+		info.SetPpEnabledLayerNames(NewCharᶜᵖᶜᵖ(layersData.Ptr()))
+		info.SetEnabledExtensionCount(uint32(len(exts)))
+		info.SetPpEnabledExtensionNames(NewCharᶜᵖᶜᵖ(extsData.Ptr()))
+		infoData := s.AllocDataOrPanic(ctx, info)
+
+		newCmd := cb.VkCreateInstance(infoData.Ptr(), cmd.PAllocator(), cmd.PInstance(), cmd.Result())
+		for _, d := range validationLayersData {
+			newCmd.AddRead(d.Data())
+		}
+		newCmd.AddRead(debugReportExtNameData.Data())
```

Comment:
I think AddRead() returns the cmd.

newCmd.AddRead().AddRead().AddRead()....

Reply:
Fixed in 2b9a068